### PR TITLE
Adding fork of xunit.console

### PR DESCRIPTION
**Pull request by josalem**
This PR adds a fork of the xunit console runner (v2.4.1) to Arcade.  The console runner for v2 of xunit is no longer being maintained, so this will provide a centralized source for long term maintenance and feature addition until a more viable solution, e.g., a backwards-compatible v3 of xunit, comes along.

The source code under `src/Microsoft.DotNet.XUnitConsoleRunner` generates a `xunit.console.netcore.dll` that can be used in the same way as the `xunit.console.runner` nuget package's `xunit.console.dll`.

#1613

### CODE REVIEW

**Comment by BruceForstall** [comment]
@josalem Is this ready to go, modulo reviews?

@ViktorHofer ping

**Comment by josalem** [comment]
As far as I'm aware, this should be good to go sans any issues brought up by review.  My understanding is that Arcade's build should handle turning this into a nupkg available on the blob feed, so all that's left is reviewing the changes.

**Comment by sergiy-k** [comment]
@ViktorHofer, @markwilkie, is there anything specific that needs to be done before this PR can be accepted? Thank you!

**Comment by ViktorHofer** [comment]
Thanks. I'll review the PR later today.

**Comment by ViktorHofer** [comment]
Quick question, why does it generate a `xunit.console.netcore.dll` and not a `xunit.console.dll`? I would expect the latter.

**Comment by josalem** [comment]
I modified the name to differentiate from the official XUnit console runner they still have hosted on NuGet.  Mainly so we are not publishing a package with the same name and being construed as the _official_ XUnit console runner.  It's a superficial change, so it would be trivial to change it back.

**Comment by ViktorHofer** [comment]
I would prefer if we keep the official assembly name and only change the package name.

**Comment by ViktorHofer** [comment]
Looking now...

**Comment by ViktorHofer** [comment]
Do we really need the netcoreapp1.0 build configuration?

File: src/Microsoft.DotNet.XUnitConsoleRunner/src/Sinks/DiagnosticMessageSink.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+                    if (!noColor)
+                        ConsoleHelper.ResetColor();
+                }
+            }
+
+            return true;
+        }
+
+#if NETFRAMEWORK
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.XUnitConsoleRunner/src/Sinks/DiagnosticMessageSink.cs]: pls remove.

File: src/Microsoft.DotNet.XUnitConsoleRunner/src/Sinks/MarshalByRefObject.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+﻿#if NETCOREAPP1_0
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.XUnitConsoleRunner/src/Sinks/MarshalByRefObject.cs]: pls remove this file if we don't need the netcoreapp1.0 build configuration.

File: src/Microsoft.DotNet.XUnitConsoleRunner/src/Utility/ConsoleLogger.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.XUnitConsoleRunner/src/Utility/ConsoleLogger.cs]: I don't see any references to this file?

**Comment by ViktorHofer** [comment]
@josalem I removed other unused code (+ partially remaining netfx ifdefs). We are still adding 5.7k lines to arcade which is the bare minimum as I believe. I'm now experimenting if I can combine the Microsoft.DotNet.XUnitRunnerUap with this package.

**Comment by ViktorHofer** [comment]
I'm done with the required work. We are now using the sources here for different variations of the runner (netcoreapp2.0, uap, uapaot), I'm very happy about that. Tested locally and everything works as expected.

**Comment by ViktorHofer** [comment]
Thanks a lot @josalem for your work!

**Comment by sergiy-k** [comment]
@ViktorHofer, @josalem, thank you so much!

**Comment by BruceForstall** [comment]
Are we now using this for corefx test runs (and others)?

cc @echesakovMSFT

**Pull request merged**